Content.Server/Database/ServerDbBase.cs

```diff
@@ -505,16 +505,23 @@ public async Task<ServerBanExemptFlags> GetBanExemption(NetUserId userId)
         public async Task EditServerRoleBan(int id, string reason, NoteSeverity severity, DateTimeOffset? expiration, Guid editedBy, DateTimeOffset editedAt)
         {
             await using var db = await GetDb();
+            var banTime = await db.DbContext.RoleBan
+                .Where(b => b.Id == id)
+                .Select(b => b.BanTime)
+                .SingleOrDefaultAsync();
 
-            var ban = await db.DbContext.RoleBan.SingleOrDefaultAsync(b => b.Id == id);
-            if (ban is null)
+            if (banTime == default)
                 return;
-            ban.Severity = severity;
-            ban.Reason = reason;
-            ban.ExpirationTime = expiration?.UtcDateTime;
-            ban.LastEditedById = editedBy;
-            ban.LastEditedAt = editedAt.UtcDateTime;
-            await db.DbContext.SaveChangesAsync();
+
+            await db.DbContext.RoleBan
+                .Where(b => b.BanTime == banTime)
```

Comment:
Can you get the user out from the roleban? Seems omega-suss doing time matches.

Reply:
Ban time goes down to 6 digits of precision on the milliseconds and we hardly role ban twice a day, but I get your point added in the PlayerID to make sure it wont happen.